test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -2332,7 +2332,7 @@ BOOST_AUTO_TEST_CASE(explicit_conversion_from_decimal_to_bytesxx)
 {
 	char const* text = R"(
 		contract Foo {
-			bytes32 a = bytes32(7);
```

Comment:
Please extract the test into a new syntax test.

Reply:
And actually it would be better to not change the source code but rather see that it will report an error.

Reply:
Actually, I see no reason that this explicit conversion (rational -> fixed bytes) shouldn't be allowed. This applies to the other tests, too. This is similar to #3796 and #3799.

Reply:
How does `bytes3 x = 0xabcdef` work?

Reply:
Because the rational's `mobileType` is `uint24`, so it is implicitly convertible. But in other cases, that doesn't work.

Reply:
OK, I see. We should still check whether we break anything there, especially with regards to leading zeros. @axic - what do you think?